Il2CppInterop.Runtime/Injection/Hook.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using Il2CppInterop.Common;
+using Microsoft.Extensions.Logging;
+
+namespace Il2CppInterop.Runtime.Injection
+{
+    public abstract class Hook<T> where T : Delegate
+    {
+        private bool isApplied;
+        private T detour;
+        internal T method;
+        internal T original;
```

Comment:
Wrong naming, fields should be private and begin with `_` and `_original` should be exposed with something like
```cs
public T Original => _original;
```